#!/usr/bin/env python

import rospkg

PACKAGE = 'coral_usb'

from dynamic_reconfigure.parameter_generator_catkin import *

rospack = rospkg.RosPack()
pkg_path = rospack.get_path('coral_usb')
model_file = os.path.join(
    pkg_path,
    './models/mobilenet_ssd_v2_coco_quant_postprocess_edgetpu.tflite')
label_file = os.path.join(pkg_path, './models/coco_labels.txt')

gen = ParameterGenerator()

gen.add('score_thresh', double_t, 0, 'Threshold for confidence score', 0.60, 0.0, 1.0)
gen.add('top_k', int_t, 0, 'Threshold for confidence score', 100, 0, 1000)
gen.add('model_file', str_t, 0, 'Model file path', model_file)
gen.add('label_file', str_t, 0, 'Label file path', label_file)

exit(gen.generate(PACKAGE, PACKAGE, 'EdgeTPUObjectDetector'))
